Handle rubygems failure

stuartmg opened the pull request:
This PR rescues any errors that occur while fetching the gem info from rubygems.org. This information is only used to create links to the gem homepage, documentation, and source. An error during this lookup should not prevent the report from being generated.

This also updates the version of a number of dependencies so that it can be more easily used by other projects.

Finally, this removes the `Gemfile.lock` file from the repository and ignores it.

This PR addresses #35 and #36.

stuartmg committed:
Removing Gemfile.lock and ignoring it

stuartmg committed:
Upgrading dependencies

stuartmg committed:
Ignoring failure to retreive gem info - only needed for report links

magneland commented on gemsurance.gemspec:
```diff
@@ -19,12 +19,12 @@ Gem::Specification.new do |s|
   s.executables               = ["gemsurance"]
 
-  s.add_dependency("bundler", "~> 1.2")
+  s.add_dependency("bundler", ">= 1.2")
```
Could you revert this line? The master branch was updated to handle this requirement (with a `< 3.0`).

magneland commented on gemsurance.gemspec:
```diff
@@ -19,12 +19,12 @@ Gem::Specification.new do |s|
-  s.add_development_dependency("rake", "0.9.2.2")
-  s.add_development_dependency("nokogiri", "1.10.0")
-  s.add_development_dependency("test-unit", "3.0.9")
+  s.add_development_dependency("mocha", "1.9.0")
```
The dev dependencies got moved into Gemfile. If you want to update them there, feel free.

magneland commented on gemsurance.gemspec:
```diff
@@ -19,12 +19,12 @@ Gem::Specification.new do |s|
-  s.add_development_dependency("nokogiri", "1.10.0")
-  s.add_development_dependency("test-unit", "3.0.9")
+  s.add_development_dependency("mocha", "1.9.0")
+  s.add_development_dependency("rake", ">= 10.0.0")
```
I think this should be OK. I doubt we make use of any old rake features.
Although note what's in [gem_info_retriever_test.rb](https://github.com/appfolio/gemsurance/tree/master/test/unit/gem_info_retriever_test.rb)

magneland commented on lib/gemsurance/gem_info_retriever.rb:
```diff
@@ -148,5 +148,13 @@ def retrieve(options = {})
+    def get_gem_info(gem_name)
+      Gems.info(gem_name)
+    rescue StandardError
+      {}
```
OK, I did a little digging here. I suppose the idea is to guard against any network errors here? 
https://github.com/rubygems/gems/blob/v0.8.0/lib/gems/request.rb
The newer versions will have a check for JSON parsing errors here:
https://github.com/rubygems/gems/pull/41
I don't know which errors you are experiencing here?
In any case, could you output a warning here?
Something like:
```
rescue StandardError => e
  $stderr.puts e
  {}
end
```

magneland reviewed (changes requested):
Thank you so much for this Pull Request, Stuart!
Sorry about being a bit slow to respond to these.

ansh-wal committed:
Bugfix
- Fix issue with index method because of the bundler gem update.
- Index method no longer directly accessible like this so update the gem
  according to the new method specs.

markwilliamson503 committed:
Merge pull request #2 from patientslikeme/bug-fix/bundler-version-upgrade

Bugfix Bunlder Update

markwilliamson503 committed:
Adding GHSA exceptoin capability.

markwilliamson503 committed:
Fix bug
